### PR TITLE
chore: (litmus-portal): Updated Redux, Community Page, SideBar and BreadCrumbs. Added Header, Notification Dropdown, Profile Dropdown.

### LINKED ISSUE #1571: Header, Notifications and Profile Section Dev.
Header, Notifications and Profile Section development for Litmus Portal.

### PR DESCRIPTION
**Pull request by ishangupta-ds**
What this PR does / why we need it:

Updated Redux, Community Page, SideBar and BreadCrumbs. Added profile icon, time series plots for community page, header, profile drop-down and notifications drop-down.

Which issue this PR fixes:

fixes #1571
fixes partially #1607 

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by gdsoumya** [comment]
Resolve Conflicts and Sing-off your commits.

**Comment by gdsoumya** [comment]
@ishangupta-ds there are no analytics data in the homepage, it was previously there.

**Comment by ishangupta-ds** [comment]
> @ishangupta-ds there are no analytics data in the homepage, it was previously there.

Yeah i am sending PR to litmuschaos/charthub.litmuschaos.io for updating the analytics API, this is happening because of the redux update which is not compatible with the current analytics API.

After that it should work fine.

File: litmus-portal/frontend/src/components/CommunityTimeSeriesGrowthPlot/index.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+          label: '1y',
+        },
+        {
+          step: 'all',
+          label: 'default',
+        },
+      ],
+    };
+
+    const { x } = this.state;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/CommunityTimeSeriesGrowthPlot/index.tsx:125]: destructure all variables at once...

```suggestion
    const {x, y, y2} = this.state;
```

File: litmus-portal/frontend/src/components/CommunityTimeSeriesTrendPlot/index.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    const y2: string[] = [];
+    rawData.forEach(function dateSplit(datum: any) {
+      const splits = datum.date.split('-');
+      x.push(
+        new Date(splits[0] as any, (splits[1] as any) - 1, splits[2] as any)
+      );
+      y.push(datum.operatorInstalls);
+      y2.push(datum.experimentRuns);
+    });
+    this.setState({ x });
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/CommunityTimeSeriesTrendPlot/index.tsx:58]: use one setState to update the object

**Comment by gdsoumya** [comment]
> > @ishangupta-ds there are no analytics data in the homepage, it was previously there.
> 
> Yeah i am sending PR to litmuschaos/charthub.litmuschaos.io for updating the analytics API, this is happening because of the redux update which is not compatible with the current analytics API.
> 
> After that it should work fine.

The basic analytics data should be there anyways like operator installs and experiments. Those details haven't changed

File: litmus-portal/frontend/src/components/CommunityTimeSeriesTrendPlot/index.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+          label: '1y',
+        },
+        {
+          step: 'all',
+          label: 'Default',
+        },
+      ],
+    };
+
+    const { x } = this.state;
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/CommunityTimeSeriesTrendPlot/index.tsx:109]: destructure all variables at once...

```suggestion
    const {x, y, y2} = this.state;
```

File: litmus-portal/frontend/src/components/Header/index.tsx [hunk 4]
```diff
@@ -1,20 +1,147 @@
...
+      transformOrigin={{ vertical: 'top', horizontal: 'right' }}
+      open={isMenuOpen}
+      onClose={handleMenuClose}
+    >
+      <MenuItem onClick={handleMenuClose}>Profile</MenuItem>
+      <MenuItem onClick={handleMenuClose}>My account</MenuItem>
+    </Menu>
+  );
+
+  const myCallback = () => {};
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Header/index.tsx:43]: change it to a more descriptive name

File: litmus-portal/frontend/src/components/Header/styles.ts [hunk 5]
```diff
@@ -1,10 +1,17 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import { makeStyles, Theme } from '@material-ui/core/styles';
 
 const useStyles = makeStyles((theme: Theme) => ({
   appBar: {
     width: '100%',
     height: '100%',
+    backgroundColor: '#FAFBFD',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/Header/styles.ts:7]: take this color from theme.palatte.background.default

File: litmus-portal/frontend/src/components/NotificationListItem/styles.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(() => ({
+  empty: {},
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/NotificationListItem/styles.ts:4]: remove this file if not required

**Comment by ishangupta-ds** [comment]
> > > @ishangupta-ds there are no analytics data in the homepage, it was previously there.
> > 
> > 
> > Yeah i am sending PR to litmuschaos/charthub.litmuschaos.io for updating the analytics API, this is happening because of the redux update which is not compatible with the current analytics API.
> > After that it should work fine.
> 
> The basic analytics data should be there anyways like operator installs and experiments. Those details haven't changed

yes indeed, i haven't modified files affecting that data will look into it..

File: litmus-portal/frontend/src/components/ProfileDropdown/index.tsx [hunk 7]
```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useRef, useState, useCallback } from 'react';
+import { IconButton, Grid } from '@material-ui/core';
+import Avatar from '@material-ui/core/Avatar';
+import ExpandMoreTwoToneIcon from '@material-ui/icons/ExpandMoreTwoTone';
+import Typography from '@material-ui/core/Typography';
+import ProfileInfoDropdownItems from '../ProfileDropdownItems';
+import useStyles from './styles';
+
+const ProfileDropdown = ({ callbackFromParent }: any) => {
+  localStorage.setItem('User', 'Richard Hill');
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/components/ProfileDropdown/index.tsx:10]: no need to set and get from local storage in the same component...we will add this if required...use state variable or constant variables if you want to use dummy data

File: litmus-portal/frontend/src/pages/Community/styles.ts [hunk 8]
```diff
@@ -53,6 +53,16 @@ const useStyles = makeStyles((theme) => ({
       flexDirection: 'column',
     },
   },
+  paperTimeSeries: {
+    width: theme.spacing(80),
+    height: theme.spacing(57.45),
+    color: '#FFFFFF',
```
Comment (arkajyotiMukherjee) [inline litmus-portal/frontend/src/pages/Community/styles.ts:59]: theme.palatte.common.white

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/pages/Community/styles.ts]: All changes suggested have been made please verify the same.

**Comment by ishangupta-ds** [comment]
All the changes suggested have been made please verify the same. @arkajyotiMukherjee

File: litmus-portal/frontend/src/config/index.ts [hunk 9]
```diff
@@ -4,6 +4,7 @@ export default {
     url:
       process.env.ANALYTICS_API ||
       'https://staging-hub.litmuschaos.io/api/community',
+    // 'http://localhost:8080/community',
```
[the remark this answers is not in the document]
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/config/index.ts]: Done..

**Comment by ishangupta-ds** [comment]
The changes have been made for the spellings.

**Comment by ishangupta-ds** [comment]
All changes requested by @rajdas98, @gdsoumya  and @arkajyotiMukherjee have been made. Please verify after merging the PR #200 for litmuschaos/charthub.litmuschaos.io.

**Comment by ishangupta-ds** [comment]
Further merge conflicts were resolved.

File: litmus-portal/frontend/src/redux/reducers/analytics.ts [hunk 10]
```diff
@@ -34,8 +36,13 @@ export const communityData = createReducer<CommunityData>(initialState, {
         count: c[3],
       });
     });
+<<<<<<< HEAD
```
Comment (gdsoumya) [inline litmus-portal/frontend/src/redux/reducers/analytics.ts:39]: remove these

File: litmus-portal/frontend/src/redux/reducers/analytics.ts [hunk 11]
```diff
@@ -34,8 +36,13 @@ export const communityData = createReducer<CommunityData>(initialState, {
...
 
+    const { geoCountry } = data.google;
```
Comment (gdsoumya) [inline litmus-portal/frontend/src/redux/reducers/analytics.ts:41]: This is not required

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/redux/reducers/analytics.ts]: removed..

File: litmus-portal/frontend/src/redux/reducers/analytics.ts (hunk 10 above)
Comment (ishangupta-ds) [reply litmus-portal/frontend/src/redux/reducers/analytics.ts]: removed..

File: litmus-portal/frontend/src/redux/reducers/analytics.ts (hunk 11 above)
Comment (gdsoumya) [reply litmus-portal/frontend/src/redux/reducers/analytics.ts:41]: It's still there. I think your commits had some problem maybe.

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/redux/reducers/analytics.ts]: yeah..now its removed..

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/redux/reducers/analytics.ts]: These changes have been made please verify @rajdas98 @gdsoumya and @arkajyotiMukherjee .

File: litmus-portal/frontend/src/models/analytics.ts [hunk 12]
```diff
@@ -1,44 +1,91 @@
...
+        dailyExperimentData,
+        dailyOperatorData,
+        monthlyExperimentData,
+        monthlyOperatorData,
+      },
+    };
+  },
+});
+
+export default communityData;
```
Comment (gdsoumya) [inline litmus-portal/frontend/src/models/analytics.ts:91]: This should be in the reducer folder not in the model. You have removed the models file, this won't build. Please run the code and then send PRs. cc: @arkajyotiMukherjee

Comment (ishangupta-ds) [reply litmus-portal/frontend/src/models/analytics.ts]: Yeah the bug is now fixed. @gdsoumya and @arkajyotiMukherjee .

**Comment by ishangupta-ds** [comment]
> Remove .DS_Store files

Done @gdsoumya

**Comment by gdsoumya** [comment]
> > Remove .DS_Store files
> 
> Done @gdsoumya

It's still there

**Comment by ishangupta-ds** [comment]
> > > Remove .DS_Store files
> > 
> > 
> > Done @gdsoumya
> 
> It's still there

can you point to the file?

**Comment by ishangupta-ds** [comment]
> > > Remove .DS_Store files
> > 
> > 
> > Done @gdsoumya
> 
> It's still there

I have added it to .gitignore actually those were from a previous commit which have been rolled back and i force pushed updated one on top of it. It wont add the .DS_Store file to litmuschaos/litmus repo.

**Comment by gdsoumya** [comment]
> > > > Remove .DS_Store files
> > > 
> > > 
> > > Done @gdsoumya
> > 
> > 
> > It's still there
> 
> I have added it to .gitignore actually those were from a previous commit which have been rolled back and i force pushed updated one on top of it. It wont add the .DS_Store file to litmuschaos/litmus repo.

https://github.com/ishangupta-ds/litmus/blob/litmus-portal/litmus-portal/.DS_Store @ishangupta-ds

**Comment by ishangupta-ds** [comment]
> > > > > Remove .DS_Store files
> > > > 
> > > > 
> > > > Done @gdsoumya
> > > 
> > > 
> > > It's still there
> > 
> > 
> > I have added it to .gitignore actually those were from a previous commit which have been rolled back and i force pushed updated one on top of it. It wont add the .DS_Store file to litmuschaos/litmus repo.
> 
> https://github.com/ishangupta-ds/litmus/blob/litmus-portal/litmus-portal/.DS_Store @ishangupta-ds

can you check again now?

**Comment by ishangupta-ds** [comment]
Deleted .DS_Store files. @gdsoumya

**Pull request closed without merging**